### ISSUE TITLE
Langevin dynamics crashes with "Killed: 9" if system is poorly minimized

### ISSUE THREAD

**Issue by avirshup** [openmm/openmm#1494]
Some of my Langevin dynamics runs have been dying without any error output - they just print out `Killed: 9`, and then terminate. It turns out that this happens if I fail to run `simulation.minimizeEnergy()` before starting dynamics. Which makes sense! But a helpful error message or even a recoverable error would be great.

**Reproducing:**
To reproduce, please see the attached example [KilledWithSignal9.zip](https://github.com/pandegroup/openmm/files/269142/KilledWithSignal9.zip)

Here's what happens when I try to run this example:

```
[example]$ python run.py
Equilibrating...
Killed: 9
```

The error can't be caught in python - it kills the entire interpreter.

**Expected:**
An error message would be nice, an exception in python would be even nicer.

**Environment:**
I get this error using the CPU platform with python 2.7, BOTH with
- OS X 10.11.3 with OpenMM 6.3, and with
- Ubuntu 14.04.3 with OpenMM 7.0

**Comment by peastman** [comment]
My best guess is that Python is somehow getting confused.  The message "Killed: 9" indicates that someone sent it a SIGKILL signal with `kill -9 <pid>`.  But it doesn't sound like that's really happening.  I suspect it's actually a segfault, but it somehow misinterprets that as an external signal instead.

Anyway, as for what the real cause is...  I see you're using the CPU platform.  When a simulation blows up and particle coordinates become nan, there are certain situations where that can lead to a segfault in the CPU platform.  It tries hard to detect that in advance and throw an exception instead.  But it isn't perfect, and sometimes a nan manages to slip through.  I'll take a look at your example and see if there's a way to improve the checking.

If your initial conformation contains bad clashes that produce large forces, that will cause the simulation to blow up.  And doing an energy minimization first eliminates the large forces.

**Comment by peastman** [comment]
Actually it's not a segfault, but it is blowing up.  I replaced `simulation.step(100)` with the following:

```Python
for i in range(100):
    print(i)
    positions = simulation.context.getState(getPositions=True).getPositions()
    print(min(positions))
    print(max(positions))
    simulation.step(1)
```

So before each step it prints out the range spanned by the coordinates of all atoms.  Here's what it prints:

```
0
(-2.34047833, 0.14160927, 1.26300973) nm
(3.5165, -1.6665, 1.8913000000000002) nm
1
(-2.341707474706881, 0.14048655967538937, 1.2650436439968153) nm
(3.5131725303585446, -1.668052456276793, 1.8961447106865719) nm
2
(-33504.00052545612, 6087.880228443673, 29565.07667463748) nm
(5313.007285085142, 12381.682266317519, -32864.56682054327) nm
```

So after two time steps the system explodes and you get atoms with ridiculously large coordinate values.  At this point it becomes significant that you specified `nonbondedMethod=app.CutoffNonPeriodic`.  You have a cutoff, so it tries to build a neighbor list.  To do that, it first needs to build a spatial subdivision data structure.  If you had periodic boundary conditions, that would cover the volume of the periodic box.  But since you don't, it instead needs to cover the full range of atomic coordinates, which in this case is absurdly large.  It spends a few minutes trying to build that structure before eventually failing, probably from running out of memory.

**Comment by jchodera** [comment]
Is there a way to guard against these kinds of failures and instead throw an Exception? If it won't be possible to construct a pair list, it probably shouldn't even try!

**Comment by peastman** [comment]
This was fixed by #2461.
